hw/ip/prim/util/sparse-fsm-encode.py

```diff
@@ -219,6 +240,29 @@ def main():
 );
 '''.format(state_str))
 
+    elif args.language == "c":
+        print("// Encoding generated with\n"
```

Comment:
not sure, but you may also want to print out some additional guidance here to instruct the user to copy the whole comment for reproducibility / auditability (as in the RTL case)...

Reply:
I've made something like your instructions and we print it out now.